2.0.0

chrisolof opened the pull request:
Updated cookbook to allow [chef-solr](https://github.com/dwradcliffe/chef-solr) to do the heavy lifting of actually provisioning the server with Apache Solr, configuring solr at a base level, and starting the service.

This cookbook now just performs the last step of copying the drupal-specific configuration files into place within the solr instance.  Both [search_api_solr](https://www.drupal.org/project/search_api_solr) and [apachesolr](https://www.drupal.org/project/apachesolr) require this step after solr is running to get solr to run with Drupal.  Because the specific configuration files needed and their destinations within the solr instance will differ on different Drupal sites and different Solr configurations, this cookbook makes no assumptions and allows for complete customization as far as what goes where.

chrisolof committed:
Dropped most functionality to allow chef-solr cookbook to do the heavy lifting.

chrisolof committed:
Added a solr service restart after configuration files have been copied.

chrisolof committed:
Added missing quotes, fixed incorrect quote usage.

timodwhit commented on README.md:
```diff
@@ -1,73 +1,30 @@
-default['drupal_solr']['app_name']  = "solr"
-default['drupal_solr']['war_dir']   = "/opt/solr"
-default['drupal_solr']['home_dir']  = "/opt/solr/#{node['drupal_solr']['app_name']}"
+License: [MIT](https://github.com/dwradcliffe/chef-solr/blob/master/LICENSE)
```
This should point to our license.

chrisolof replied:
Oh sure enough! - fixing...

timodwhit commented:
Might want to add more information about the configuration file. Like a base solr install. Also, if this depends on chef-solr, we should probably put the recipe in the berksfile and in the default recipe.

timodwhit commented:
@arknoll what do you think about the conf files being deleted?

arknoll commented:
@chrisolof Hmm... So this is assuming you have all of the config files somewhere in your codebase? Before all of the config files were generated from templates. The benefit of this is the paths within the files were generated on provision. Now the problem is the paths within those files in your repo will likely be the paths for dev/stage/prod, not necessarily local.

What are your thoughts on this?

timodwhit commented:
Yeah, I agree. What if we still did the templating but provide an overriding of that. `"template":"false"` type of thing?

This would be similar to how we handle other config/files.

chrisolof commented:
@timodwhit I left chef-solr out because nothing in this specifically relies upon that cookbook - and there are multiple solr cookbooks that can get your solr instance in place in just the way you want.

@arknoll If you follow the installation instructions on either of the two most popular solr implementation modules on Drupal, the first step is getting your solr instance running (that can be done with chef-solr), and the second step, in both modules, is to move certain configuration files from within the contrib module into the solr instance's configuration directory.  So those files are already present in your Drupal repo - so I see no reason to maintain those files here in this cookbook as well.  I'm also not sure why your solr configuration files would differ from dev/stage/prod - I would imagine your solr endpoint would differ, and when using vampd + this cookbook your solr server endpoint would simply be on your local machine.  Maybe we're miscommunicating...

chrisolof commented:
See:
http://cgit.drupalcode.org/search_api_solr/tree/INSTALL.txt?id=7.x-1.6
and
http://cgit.drupalcode.org/apachesolr/tree/README.txt?id=7.x-1.7

timodwhit commented:
@chrisolof It would seem like a dependency on the cookbook would be good, there are multiple cookbooks to do mysql, etc but we should pick the one that suits most needs so people can get up and running reliably.

Also, I think if we are looking to copy from our repo to the server, then we should clarify in the Readme, what it would look like. Just provide an example of what the config might look like.

arknoll commented on templates/default/solr_context.xml.erb:
```diff
@@ -1,8 +0,0 @@
-  Local modifications will be overwritten by Chef
--->
-<Context docBase="<%= node['drupal_solr']['war_dir'] %>/solr.war" debug="0" crossContext="true">
-  <Environment name="solr/home" type="java.lang.String" value="<%= node['drupal_solr']['home_dir']%>" override="true" />
```
@chrisolof this specifically is what I was concerned about. node['drupal_solr']['home_dir'] could be different on local/dev/stage/prod

chrisolof committed:
Fixed incorrect license reference.

chrisolof committed:
Added a setup example with Vampd.

chrisolof commented:
@arknoll @timodwhit I've removed the incorrect license reference and added an example Vampd setup section to the readme.

Also - as neither of the popular solr modules for drupal touch the solr_context.xml file, I'm allowing that file to be setup by chef-solr.  drupal-solr is just concerned with migrating the drupal-specific config and assumes you've already got your solr instance configured and working (via chef-solr or some other route) - which would include a properly set up solr_context.xml.  In the example now provided in the readme I never had to touch that file as it came pre-configured from the apache solr installation.

Let me know if you guys think this is ready for merge or if it needs anything else (or more discussion).

Thanks!

chrisolof commented:
Bump...  Any outstanding issues here?
